### PR TITLE
⬆️ Maintenance: upgrade to python 3.9.12

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?
This upgrades the whole repository to the latest python 3.9.12 version (thus making it fully compatible with ARM/ARMv64 versions in piwheels.org)

In order to upgrade once the PR goes in:
```
make clean-all
pyenv update
pyenv install 3.9.12
pyenv local 3.9.12
make devenv
source .venv/bin/activate
make build build-devel
```

some new features:
- dict unions
- type hinting using ```dict, list```
- string method to remove prefixes and suffixes

[full reference](https://docs.python.org/3/whatsnew/3.9.html)

### CODE REVIEW

**Review by KZzizzle — COMMENTED** [review]
Thanks for the notice!

File: packages/service-library/src/servicelib/aiohttp/application_setup.py [hunk 1]
```diff
@@ -89,7 +89,7 @@ def _is_addon_enabled_from_config(
         for part in parts:
             if section and part == "enabled":
                 # if section exists, no need to explicitly enable it
-                return strtobool(f"{searched_config.get(part, True)}")
+                return parse_obj_as(bool, f"{searched_config.get(part, True)}")
```
Comment (pcrespov) [inline packages/service-library/src/servicelib/aiohttp/application_setup.py:92]: There is no need to convert to ``str`` here, it suffices with
```python
parse_obj_as(bool,  searched_config.get(part, True))
```

**Review by pcrespov — APPROVED** [review]
:tada: great to see how smoothly we transition this time! :-)

File: packages/service-library/src/servicelib/aiohttp/application_setup.py (hunk 1 above)
Comment (sanderegg) [reply packages/service-library/src/servicelib/aiohttp/application_setup.py:92]: true that.

**Review by GitHK — APPROVED** [review]
I am truly amazed by this! 🎉 
Was expecting way more changes!

**Review by mrnicegyu11 — APPROVED** [review]

**Pull request merged**